core/rfb.js

```diff
@@ -1203,6 +1223,14 @@
                     now = (new Date()).getTime();
                     Util.Info("First FBU latency: " + (now - this._timing.fbu_rt_start));
                 }
+
+                // Make sure the previous frame is fully rendered first
+                // to avoid building up an excessive queue
+                if (this._display.pending()) {
+                    this._flushing = true;
```

Comment:
you'd probably need to rewind here, since you've already consumed 3 bytes worth of information (unless I'm misreading something).